CONTRIBUTING.md

```diff
@@ -8,30 +8,59 @@ Raising [issues](https://github.com/shuttle-hq/shuttle/issues) is encouraged. We
 You can use Docker and docker-compose to test shuttle locally during development. See the [Docker install](https://docs.docker.com/get-docker/)
 and [docker-compose install](https://docs.docker.com/compose/install/) instructions if you do not have them installed already.
 
-You should now be set to run shuttle locally as follow:
+You should now be ready to setup a local environment to test code changes to core `shuttle` packages as follows:
+
+Build the required images with:
+
+```bash
+$ docker buildx bake -f docker-bake.hcl provisioner backend
+```
+
+The images get built with [cargo-chef](https://github.com/LukeMathWalker/cargo-chef) and therefore support incremental builds (most of the time). So they will be much faster to re-build after an incremental change in your code - should you wish to deploy it locally straightaway.
+
+Create a docker persistent volume with:
 
 ```bash
-# clone the repo
-git clone git@github.com:shuttle-hq/shuttle.git
+$ docker volume create shuttle-backend-vol
+```
+
+Finally, you can start a local deployment of the backend with:
+
+```bash
+$ docker compose -f docker-compose.dev.yml up -d
+```
+
+The API is now accessible on `localhost:8000` (for app proxies) and `localhost:8001` (for the control plane). When running `cargo run --bin cargo-shuttle` (in a debug build), the CLI will point itself to `localhost` for its API calls. The deployment parameters can be tweaked by changing values in the [.env](./.env) file.
 
-# cd into the repo
-cd shuttle
+In order to test local changes to the `shuttle-service` crate, you may want to add the below to a `.cargo/config.toml` file. (See [Overriding Dependencies](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html) for more)
 
-# start the shuttle services
-docker-compose up --build
+``` toml
+[patch.crates-io]
+shuttle-service = { path = "[base]/shuttle/service" }
+```
+
+Login to shuttle service in a new terminal window from the main shuttle directory:
 
-# login to shuttle service in a new terminal window
-cd path/to/shuttle/repo
+```bash
 cargo run --bin cargo-shuttle -- login --api-key "ci-test"
+```
+
+cd into one of the examples:
 
-# cd into one of the examples
+```bash
 cd examples/rocket/hello-world/
+```
+
+Deploy the example:
 
-# deploy the example
+```bash
 # the --manifest-path is used to locate the root of the shuttle workspace
 cargo run --manifest-path ../../../Cargo.toml --bin cargo-shuttle -- deploy
+```
+
+Test if the deploy is working:
 
-# test if the deploy is working
+```bash
 # (the Host header should match the Host from the deploy output)
 curl --header "Host: hello-world-rocket-app.teste.rs" localhost:8000/hello
```

Comment:
While we're at it, [this should probably be](https://github.com/shuttle-hq/shuttle/blob/036f2ac9f856a789817b6194fb73fb911abb3651/.env#L3) `{app}.localhost.local`.